File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  SOUNDS = [[3, 'Pling'],
            [5, 'Plang'],
            [7, 'Plong']].freeze

  def self.convert(num)
    result = ''
    SOUNDS.each do |(divisor, sound)|
      result << sound if (num % divisor).zero?
    end
    result == '' ? num.to_s : result
  end
end

File: csharp/raindrops/Raindrops.cs
using System;
using System.Collections.Generic;

public static class Raindrops
{
    static readonly Dictionary<int, string> Sounds = new Dictionary<int, string>()
    {
        {3, "Pling"},
        {5, "Plang"},
        {7, "Plong"}
    };
            
    public static string Convert(int number)
    {
        string result = "";
        
        foreach (KeyValuePair<int, string> sound in Sounds) {
            result += number % sound.Key == 0 ? sound.Value : "";
        }
        
        return String.IsNullOrEmpty(result) ? number.ToString() : result;
    }
}